From ec739861f0f7b82bb0a70da74d79c3daf356f71a Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Thu, 13 Oct 2011 09:55:24 +0000
Subject: shell: Make Ctrl+W close panels

Ctrl+W will close the current panel, and go back to the
overview.

https://bugzilla.gnome.org/show_bug.cgi?id=660814
---
diff --git a/shell/gnome-control-center.c b/shell/gnome-control-center.c
index 42fb47b..3f1e211 100644
--- a/shell/gnome-control-center.c
+++ b/shell/gnome-control-center.c
@@ -1055,6 +1055,12 @@ window_key_press_event (GtkWidget          *win,
             g_object_unref (self);
             retval = TRUE;
             break;
+          case GDK_KEY_W:
+          case GDK_KEY_w:
+            if (gtk_notebook_get_current_page (GTK_NOTEBOOK (self->priv->notebook)) != OVERVIEW_PAGE)
+              shell_show_overview_page (self->priv);
+            retval = TRUE;
+            break;
         }
     }
   return retval;
--
cgit v0.9.0.2

